src/Kdyby/DoctrineMoney/DoctrineMoney.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/**
+ * This file is part of the Kdyby (http://www.kdyby.org)
+ *
+ * Copyright (c) 2008 Filip Procházka (filip@prochazka.su)
+ *
+ * For the full copyright and license information, please view the file license.txt that was distributed with this source code.
+ */
+
+namespace Kdyby\DoctrineMoney;
+
+use Kdyby\Money\Currency;
+use Kdyby\Money\Money;
+
+
+
+/**
+ * @author Martin Štekl <martin.stekl@gmail.com>
+ */
+class DoctrineMoney extends Money implements IVariableCurrency
+{
+
+	/**
+	 * @inheritdoc
+	 */
+	public function setCurrency(Currency $currency)
```

Comment:
Pro vnější použití také je. Tohle je jen pro vnitřní potřeby knihovny, aby fungovala efektivně a dle očekávání.